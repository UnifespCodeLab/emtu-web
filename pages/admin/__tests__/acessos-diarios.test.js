import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'

import AdminDailyAccess from '~/pages/admin/acessos-diarios.vue'

const localVue = createLocalVue()
let vuetify
let wrapper

describe('Pages / Admin / AdminDailyAccess', () => {
  beforeEach(() => {
    jest.mock('vue-apexcharts', () => {})

    vuetify = new Vuetify()

    wrapper = mount(AdminDailyAccess, {
      localVue,
      vuetify,
      stubs: {
        'router-link': RouterLinkStub
      }
    })
  })

  it('should render the page', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the correct elements', () => {
    const dialogs = wrapper.findAllComponents({ name: 'v-dialog' })
    expect(dialogs.length).toBe(2)

    const textFields = wrapper.findAllComponents({ name: 'v-text-field' })
    expect(textFields.length).toBe(2)

    const datePickers = wrapper.findAllComponents({ name: 'v-date-picker' })
    expect(datePickers.length).toBe(0)

    const searchButton = wrapper.findComponent({ name: 'v-btn' })
    expect(searchButton.props('color')).toBe('primary')
  })

  describe('when the start date input is clicked', () => {
    beforeEach(() => {
      const textField = wrapper.findAllComponents({ name: 'v-text-field' }).at(0)
      textField.trigger('click')
    })

    it('should show the picker', async () => {
      await wrapper.vm.$nextTick()
      const datePickers = wrapper.findAllComponents({ name: 'v-date-picker' })
      expect(datePickers.length).toBe(1)
    })
  })
})
