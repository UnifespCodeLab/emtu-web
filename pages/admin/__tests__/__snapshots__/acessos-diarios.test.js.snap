// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages / Admin / AdminDailyAccess should render the page 1`] = `
<div
  class="daily-access"
>
  <div
    class="daily-access__time-container"
  >
    <div
      class="v-input daily-acces__input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-text-field__slot"
          >
            <input
              aria-expanded="false"
              aria-haspopup="true"
              id="input-3"
              role="button"
              type="text"
            />
          </div>
          <div
            class="v-input__append-inner"
          >
            <div
              class="v-input__icon v-input__icon--clear"
            >
              <button
                aria-label="clear icon"
                class="v-icon notranslate v-icon--link mdi mdi-close theme--light"
                type="button"
              />
            </div>
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="v-dialog__container"
    >
      <!---->
    </div>
     
    <div
      class="v-dialog__container"
    >
      <!---->
    </div>
    <div
      class="v-input daily-acces__input v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-text-field__slot"
          >
            <input
              aria-expanded="false"
              aria-haspopup="true"
              id="input-8"
              readonly="readonly"
              role="button"
              type="text"
            />
          </div>
          <div
            class="v-input__append-inner"
          >
            <div
              class="v-input__icon v-input__icon--clear"
            >
              <button
                aria-label="clear icon"
                class="v-icon notranslate v-icon--link mdi mdi-close theme--light"
                type="button"
              />
            </div>
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <button
      class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--large primary"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        
      BUSCAR
    
      </span>
    </button>
  </div>
   
  <div
    class="daily-access__page-wrapper"
  >
    <!---->
  </div>
</div>
`;
